velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -319,4 +319,20 @@ struct DayOfWeekFunction : public InitSessionTimezone<T>,
   }
 };
 
+template <typename T>
+struct DateDiffFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      int32_t& result,
+      const arg_type<Date>& date1,
+      const arg_type<Date>& date2) {
+    int64_t value = diffDate(DateTimeUnit::kDay, date2, date1);
```

Comment:
In Spark, the result is positive if the first arg is larger than the second one (
[datetimeExpressions.scala](https://github.com/apache/spark/blob/branch-3.2/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L2280-L2282C1)). But with above code, the sign of result is reverted as [DateTimeImpl.h#L198](https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/DateTimeImpl.h#L198) shows. We need to use diffDate(DateTimeUnit::kDay, date1, date2) instead.

Reply:
Just fixed.

Reply:
Thanks!